.pre-commit-config.yaml

```diff
@@ -52,6 +52,31 @@ repos:
       - id: black
         args: [--safe, --quiet]
         exclude: *fixtures
+  - repo: https://github.com/psf/black
+    rev: 23.3.0
+    hooks:
+      - id: black
+        name: black-doc
+        # Should be in sync with doc/data/ruff.toml
+        args: [--safe, --quiet, -l 103]
+        files: doc/data/messages/
+        exclude: |
```

Comment:
Can't we re-use this like we do with `&fixtures` and pass it in the `pre-commit` config?

Reply:
What need to be in sync is the ``-l 103``, black only looks at ``pyproject.toml`` so we have to give the option like this. (Alternatively we could use the default value, it's not super important.).

I don't think the exclude can be reused, this is the list of the message that black can fix automatically, it's really specific to black.

It's also a very useful list btw ! I'm considering to recover it automatically and generalize it to autofixxer (isort, autoflake, and ruff), in order to use this information directly in our doc for #8579

Reply:
I.e. if applying a tool on ``doc/data`` makes the tests fail on a ``m/my-message/bad.py`` file, then the tool autofix ``my-message``. We might have to use it on functional tests instead for complicated fixes like ruff or autoflake to check which use cases is autofixed or not, but for black on the doc is enough and you get the idea.

Reply:
Yeah, let's use the default then. Less config == better imo

Reply:
I'm going to need to rebase and rewrite black from zero, it's not THAT good with comment after all 😄

Reply:
Shall we merge this and do another PR that removes the config?

Reply:
No, it breaks the comments to go back to 88 char per line after doing 103 char per line, it's a lot easier to do it from scratch again :)